### PR TITLE
[profiler] Some documentation improvements

### PR DESCRIPTION
**Pull request by alexrp**
* Update and rewrite profiler-related `man` pages.
* Update profiler API documentation for the new API.
* Extend the `exdoc` tool to support some more Doxygen features.
* Document the AOT profiler's file format.
* Clean up the sample profiler module and add a makefile.
* Include context/domain reports in `mprof-report`'s default report set.
* Exit the process when the `help` option is passed to the profilers.
* Remove the IOMap profiler.
* Remove some deprecated options from the log profiler's `help` output.
* Clean up the AOT profiler and bring its user interface in line with the others.
* Allow comments in coverage filter files.

### CODE REVIEW

File: man/mono-profilers.1 [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+GC handle creation and deletion events.
+.TP
+\fBfinalization\fR
+Object finalization events.
+.TP
+\fBcounter\fR
+Performance counter sample events.
+.TP
+\fBjit\fR
+JIT code buffer events.
```
[the remark this answers is not in the document]
Comment (alexrp) [reply man/mono-profilers.1]: These: https://github.com/mono/mono/blob/master/mono/metadata/profiler.h#L355

I.e. not super useful to the average user (to this day I'm not exactly sure why these events were even added; it was done before I started maintaining the profiler).

File: man/mono-profilers.1 [hunk 2]
```diff
@@ -0,0 +1,299 @@
...
+exception events and performance counter sampling by default. It also
+makes it so that GC move events won't be enabled by default when
+heapshots are enabled. To use this option, it must be the first
+option given to the log profiler.
+.IP
+This option will be the default in a future version of the log
+profiler.
+.TP
+[\fBno\fR]\fIevent\fR
+Enable or disable gathering data for \fIevent\fR, which can be one
```
[the remark this answers is not in the document]
Comment (alexrp) [reply man/mono-profilers.1]: I will add a note to the ones that are enabled by default.

File: mono/profiler/log-args.c [hunk 3]
```diff
@@ -347,4 +341,6 @@ usage (void)
 	mono_profiler_printf ("\treport               create a report instead of writing the raw data to a file");
 	mono_profiler_printf ("\tzip                  compress the output data");
 	mono_profiler_printf ("\tport=PORTNUM         use PORTNUM for the listening command server");
+
+	exit (code);
```
Comment (luhenry) [inline mono/profiler/log-args.c]: How is this going to interact with us passing arguments for the old profiler API? We will need a period of transition where both command-line arguments will be in use, so I am not sure exiting when passing an unknown argument is going to help us in that.

**Comment by alexrp** [comment]
@luhenry @marek-safar the IOMap profiler has not even been functional since 41b8883053f79cc4ee08b4515297a2e422b9efa9 (2 years ago). As in, it will just fail to load because of symbol resolution issues. Considering we have not gotten a single bug report in those 2+ years, I wouldn't worry too much.

File: mono/profiler/log-args.c (hunk 3 above)
Comment (alexrp) [reply mono/profiler/log-args.c]: Good point, we should not exit on unknown arguments.

Comment (alexrp) [reply mono/profiler/log-args.c]: Actually, it's fine: In the log profiler, we just print an error if we can't parse an argument; we don't invoke `usage ()`.

**Comment by alexrp** [comment]
Added some more commits, addressed review comments, and expanded the `man` pages a bit (e.g. documenting coverage filter files).

**Comment by alexrp** [comment]
@monojenkins merge

**Pull request merged**